src/brsTypes/components/RoSGNode.ts

```diff
@@ -659,6 +709,279 @@ export class RoSGNode extends BrsComponent implements BrsValue, BrsIterable {
         },
     });
 
+    /**
+     * If the subject node has a child node in the index position, replace that child
+     * node with the newChild node in the subject node list of children, otherwise do nothing.
+     */
+
+    private replacechild = new Callable("replacechild", {
+        signature: {
+            args: [
+                new StdlibArgument("newchild", ValueKind.Dynamic),
+                new StdlibArgument("index", ValueKind.Int32),
+            ],
+            returns: ValueKind.Boolean,
+        },
+        impl: (interpreter: Interpreter, newchild: BrsType, index: Int32) => {
+            return BrsBoolean.from(this.replaceChildAtIndex(newchild, index));
+        },
+    });
+
+    /**
+     * Removes the child nodes specified by child_nodes from the subject node. Returns
+     * true if the child nodes were successfully removed.
+     */
+    private removechildren = new Callable("removechildren", {
+        signature: {
+            args: [new StdlibArgument("child_nodes", ValueKind.Dynamic)],
+            returns: ValueKind.Boolean,
+        },
+        impl: (interpreter: Interpreter, child_nodes: BrsType) => {
+            if (child_nodes instanceof RoArray) {
+                let childNodesElements = child_nodes.getElements();
+                if (childNodesElements.length != 0) {
+                    childNodesElements.forEach(childNode => {
+                        this.removeChildByReference(childNode);
+                    });
+                    return BrsBoolean.True;
+                }
+            }
+            return BrsBoolean.False;
+        },
+    });
+
+    /**
+     * Removes the number of child nodes specified by num_children from the subject node
+     * starting at the position specified by index.
+     */
+    private removechildrenindex = new Callable("removechildrenindex", {
```

Comment:
does this function have a `-1` num_children shortcut like `getChildren` function has?

Reply:
Nope, `-1` just returns `false`